libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1077,32 +1169,61 @@ BaseType_t xDoStore = xExpected;
 		{
 			for( x = 0; x < pxDNSMessageHeader->usAnswers; x++ )
 			{
-				pucByte = prvSkipNameField( pucByte,
-											uxSourceBytesRemaining );
+			BaseType_t xDoAccept;
+
+				uxResult = prvSkipNameField( pucByte,
+											 uxSourceBytesRemaining );
 
 				/* Check for a malformed response. */
-				if( NULL == pucByte )
+				if( uxResult == 0U )
 				{
-					return dnsPARSE_ERROR;
+					return dnsPARSE_ERROR;/*lint !e904 */
 				}
 
-				uxSourceBytesRemaining = ( size_t )
-					( pucUDPPayloadBuffer + uxBufferLength - pucByte );
+				/* MISRA c 2012 rule 10.8 relaxed. All the variables
+				are capable of handling the data. The should not be
+				any overflow. */
+				uxBytesRead += uxResult;
+				pucByte = &( pucByte[ uxResult ] );
+				uxSourceBytesRemaining -= uxResult;
 
 				/* Is there enough data for an IPv4 A record answer and, if so,
 				is this an A record? */
-				if( ( uxSourceBytesRemaining >= ( sizeof( DNSAnswerRecord_t ) + sizeof( uint32_t ) ) ) &&
-					( usChar2u16( pucByte ) == dnsTYPE_A_HOST ) )
+				usType = usChar2u16( pucByte );
+
+				if( usType == ( uint16_t ) dnsTYPE_A_HOST )
```

Comment:
Definition of usType is being controlled by #if( ipconfigUSE_LLMNR == 1 ) on line 1061. Causes a build failure in WinSim project.

Reply:
I just made a commit in which the variable `usType` is also defined without the use of LLMNR.

Reply:
Thanks